app/services/sorting/PostSorter/HashtagPostSorter.java

```diff
@@ -0,0 +1,113 @@
+package services.sorting.PostSorter;
+
+import services.dataAccess.AbstractDataAccess;
+import services.dataAccess.proto.PostProto.Post;
+import services.sorting.Calculator;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class HashtagPostSorter extends AbstractPostSorter {
+
+    private static final int POST_THRESHOLD = AbstractDataAccess.getMaxPostlists() / 2;
+    private static final String NO_HASHTAGS = "N/A";
+    private Calculator calc;
+
+    public HashtagPostSorter(AbstractDataAccess dataSource) {
+        super(dataSource);
+        calc = new Calculator();
+    }
+
+    /**
+     * Sorts a list of posts into individual binned lists, according to the hashtags therein
+     *
+     * @param posts list of posts to be sorted
+     * @return map of hashtag strings to lists of posts that contain the key string hashtag
+     */
+    @Override
+    public Map<String, List<Post>> sort(List<Post> posts) {
+        Map<String, List<Post>> postsByHashTag = new HashMap<>();
+
+        // sort by popularity, in decreasing order
+        List<Post> listOfPostsClone = new ArrayList<>(posts);
+        listOfPostsClone.stream()
+                .sorted(Collections.reverseOrder(Comparator.comparingInt(Post::getPopularityScore)))
+                .collect(Collectors.toList());
```

Comment:
`.collect(Collectors.toList)` is being used but the result isn't being placed anywhere.

Also this seems to be the only place listsOfPostsClone is being used. What is its purpose?